integration/kubernetes/confidential/agent_image.bats

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env bats
+# Copyright (c) 2022 Red Hat
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/lib.sh"
+load "${BATS_TEST_DIRNAME}/../../confidential/lib.sh"
+
+# Currently the agent can only check images signature if using skopeo.
+# There isn't a way to probe the agent to determine if skopeo is present
+# or not, so we need to rely on build variables. If we are running under
+# CI then we assume the variables are properly exported, otherwise we
+# should skip testing.
+#
+skip_if_skopeo_not_present () {
+	if [ "${CI:-}" == "true" ]; then
+		if [ "${SKOPEO:-no}" == "no" ]; then
+			skip "Skopeo seems not installed in guest"
+		fi
+	else
+		skip "Cannot determine skopeo is installed in guest"
+	fi
+}
+
+# Create the test pod.
+#
+# Note: the global $sandbox_name, $pod_config should be set
+# 	already. It also relies on $CI and $DEBUG exported by CI scripts or
+# 	the developer, to decide how to set debug flags.
+#
+create_test_pod() {
+	# On CI mode we only want to enable the agent debug for the case of
+	# the test failure to obtain logs.
+	if [ "${CI:-}" == "true" ]; then
+		enable_full_debug
+	elif [ "${DEBUG:-}" == "true" ]; then
+		enable_full_debug
+		enable_agent_console
+	fi
+
+	echo "Create the test sandbox"
+	echo "Pod config is: "$pod_config
+	kubernetes_create_cc_pod $pod_config
+}
+
+assert_pod_fail() {
+	local container_config="$1"
+	echo "In assert_pod_fail: "$container_config
+
+	echo "Attempt to create the container but it should fail"
+	! kubernetes_create_cc_pod "$container_config" || /bin/false
+}
+
+setup() {
+	start_date=$(date +"%Y-%m-%d %H:%M:%S")
+
+	sandbox_name="busybox-cc"
+	pod_config="${FIXTURES_DIR}/pod-config.yaml"
+	pod_id=""
+
+	echo "Delete any existing ${sandbox_name} pod"
+	kubernetes_delete_cc_pod_if_exists "$sandbox_name"
+
+	echo "Prepare containerd for Confidential Container"
+	SAVED_CONTAINERD_CONF_FILE="/etc/containerd/config.toml.$$"
+	configure_cc_containerd "$SAVED_CONTAINERD_CONF_FILE"
+
+
+	echo "Reconfigure Kata Containers"
+	switch_image_service_offload on
+	clear_kernel_params
+}
+
+# Check the logged messages on host have a given message.
+# Parameters:
+#      $1 - the message
+#
+# Note: get the logs since the global $start_date.
+#
+assert_logs_contain() {
+	local message="$1"
+    journalctl -x -t kata --since "$start_date" | grep "$message"
+}
+
+@test "[cc][agent][cri][containerd] Test can pull an unencrypted image inside the guest" {
```

Comment:
All the tests start with `[cc][agent][cri][containerd] ` so how about creating a variable to avoid the duplication? Something like:

```bash

test_tag='[cc][agent][cri][containerd]'

# ...

@test "${test_tag} Test can pull an unencrypted image inside the guest" {
```

Reply:
We've now added a `test_tag` variable as you suggested that is being passed in for each test.